src/main/java/com/openshift/internal/restclient/capability/resources/OpenShiftBinaryPortForwarding.java

```diff
@@ -0,0 +1,136 @@
+package com.openshift.internal.restclient.capability.resources;
+
+import java.io.IOException;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.authorization.BasicAuthorizationStrategy;
+import com.openshift.restclient.authorization.IAuthorizationStrategyVisitor;
+import com.openshift.restclient.authorization.KerbrosBrokerAuthorizationStrategy;
+import com.openshift.restclient.authorization.TokenAuthorizationStrategy;
+import com.openshift.restclient.capability.resources.IPortForwardable;
+import com.openshift.restclient.capability.resources.LocationNotFoundException;
+import com.openshift.restclient.model.IPod;
+
+public class OpenShiftBinaryPortForwarding implements IPortForwardable {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(IPortForwardable.class);
+	
+	private IPod pod;
+	private Process process;
+
+	private PortPair[] pairs;
+
+	private IClient client;
+
+	public OpenShiftBinaryPortForwarding(IPod pod, IClient client) {
+		this.pod = pod;
+		this.client = client;
+		Runnable runnable = new Runnable() {
+			@Override
+			public void run() {
+				stop();
+			}
+		};
+		Runtime.getRuntime().addShutdownHook(new Thread(runnable));
+	}
+	
+	
+	@Override
+	public boolean isForwarding() {
+		return process != null && process.isAlive();
+	}
+
+
+	@Override
+	public boolean isSupported() {
+		return true;
+	}
+
+	@Override
+	public String getName() {
+		return OpenShiftBinaryPortForwarding.class.getSimpleName();
+	}
+
+	@Override
+	public PortPair[] getPortPairs() {
+		return pairs;
+	}
+
+	@Override
+	public synchronized void forwardPorts(PortPair... pairs) {
+		stop();
+		if(pairs.length == 0) return;
+		String location = System.getProperty(IPortForwardable.OPENSHIFT_BINARY_LOCATION);
+		if(StringUtils.isBlank(location)) {
+			throw new LocationNotFoundException(String.format("The OpenShift 'oc' binary location was not specified. Set the property %s", IPortForwardable.OPENSHIFT_BINARY_LOCATION));
+		}
+		StringBuilder args = new StringBuilder(location);
```

Comment:
I would extract the path building to a separate method, that would keep this method focused and enhance readability